src/lib/logger.js

```diff
@@ -29,17 +29,22 @@ function getlvl(x) {
  * @param {*} logs list of log configuration
  */
 function setup(logs) {
+	let files = false;
+
 	let streams = [];
 	if (logs == null) {
 		logs = [{type: 'stdout', format: 'pretty', level: 'http'}];
 	}
 
 	logs.forEach(function(target) {
-		// create a stream for each log configuration
-		const stream = new Stream();
-		stream.writable = true;
+
+		if (target.level === undefined || target.level === 'http') {
+			target.level = 35;
+		}
 
 		if (target.type === 'stdout' || target.type === 'stderr') {
+			const stream = new Stream();
```

Comment:
If you create the `Stream` here and the `if` is false the `else` condition will fail due `stream` is undefined.
```
	} else if (target.type === 'file') {
			files = true;
			streams.push({
				type: 'file',
				level: target.level,
				path: target.path,
			});
		}
```